### ISSUE TITLE
Problem with feed

### ISSUE THREAD

**Issue by toyish** [PHP-on-Couch/PHP-on-Couch#61]
I can't get the example "Continuous changes with a callback function" in the docs page to work.
In the docs it says the following about feed,
PHPOnCouch\CouchClient::feed(string $value, $callback)
The callable function or method will receive two arguments : the JSON object decoded as a PHP object, and a couchClient instance.

_It doesn't return a couchClient but a CouchHttpAdapterSocket._

And the example from the docs:
Example - Continuous changes with a callback function

function index_doc($change,$couch) {
    if( $change->deleted == true ) {
        // won't index a deleted file
        return ;
    }
    echo "indexing ".$change->id."\n";
    $doc = $couch->getDoc($change->id);
    unset($doc->_rev);
    $id = $doc->_id;
    unset($doc->_id);
    my_super_fulltext_search_appliance::index($id, $doc);
}
$client->feed('continuous','index_doc')->getChanges();

_That $couch inside the function are not a couchClient but a CouchHttpAdapterSocket._

Is there any working example some place or some other information that can get me in the right direction?

**Comment by popojargo** [comment]
It seems like there was a regression when I added the adapters. I'll fix that and update the documentation.

**Comment by popojargo** [comment]
#62 will be merged soon. The second argument of the callback should be the couchclient object.

**Comment by popojargo** [comment]
You can download the version from dev-master. Please let me know if it fixes your problem

**Comment by popojargo** [comment]
So the library has not been updated yet. In order to get my fix, you must download the master with this:

`composer require php-on-couch/php-on-couch:dev-master`

Here's the example I just tried with success: 

```
$client = new \PHPOnCouch\CouchClient('http://localhost:5984', 'local');

function index_doc($change, $couch)
{
    if ($change->deleted == true) {
        return;
    }
    echo "indexing " . $change->id . "\n";
    $doc = $couch->getDoc($change->id);
    var_dump($doc);
}

$client->feed('continuous', 'index_doc')->getChanges();
```

**Comment by popojargo** [comment]
Once you confirm me that it's fixed, I'll create an official minor release.

**Comment by toyish** [comment]
Ok, now the second argument is a couchclient object.
But now I can't get any changes, maybee I doing it wrong?

**Comment by popojargo** [comment]
Can you try on a fresh database? Start you script and add documents. They should pass through the changes callback. 

I would recommend that you only test the functions seperatly from your app. From that, we can know if it's a code problem or an environment problem.

**Comment by toyish** [comment]
Already did that.
I changed in my .env file from socket to curl and now at least I get the page to not "finish loading", but still no new documents from the change feed.
The new documents comes after a while, like that are some timeout and then all documents at once.
Tried with longpoll but blank page, maybe missing some options for longpoll.
My dev enviroment is Windows and Laragon with nginx and php ver 7.1.7

**Comment by popojargo** [comment]
First of all, you can stick with the cURL adapter by default. continuousQueries are always using socket since it's not supported by cURL.

Let's say you create a database and create some "changes" by creating documents.
If you start the application and get all the changes, it should automatically load the changes and process them. Is this working?

Also, if you manually get the changes, do you have any delay?

**Comment by toyish** [comment]
Ok.
And yes I created a new database and saved some new documents, I reload the server and load the page, all changes was loaded on the page but if I save any new documents in CouchDB during the page is still "loading" none of the new documents are processed or loaded into the page.
After a while it seems it times out or something because then the page is "loaded" and it spit out the new documents I saved during that time.

And I checked in the changed-feed in the database and the changes are instant.
